### PR TITLE
increase test coverage

### PR DESCRIPTION
**Pull request by gyanesh-m**

### CODE REVIEW

File: amortization.go [hunk 1]
```diff
@@ -160,9 +163,18 @@ func PlotRows(rows []Row, fileName string) error {
...
+	f, err := os.Create(fmt.Sprintf("%s.html", filePath))
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err = f.Close()
+	}()
+	if writer == nil {
+		writer = f
+	}
```
Comment (krantideep95) [inline amortization.go:175]: there's issue in this approach. first time file will be opened and set to `writer`. next time, same writer is being used which may not work. (file doesn't exist etc).
instead move this in unexported method. accept `io.Writer` as func param. in unit test, pass fake io.writer to this unexported method. from `PlotRows` pass real file io.Writer.

Comment (krantideep95) [reply amortization.go:175]: if you find better approach, open to that as well.

Comment (gyanesh-m) [reply amortization.go]: oh yes, you are right. What you suggested seems perfectly fine to me. I have added it.
There was one more approach of using https://github.com/spf13/afero. Its a fs abstraction system using which we can mock any call to the fs. But I am not inclined to use that here.

File: amortization.go [hunk 2]
```diff
@@ -117,7 +115,29 @@ func PrintRows(rows []Row) {
 }
 
 // PlotRows uses the go-echarts package to generate an interactive plot from the Rows array.
-func PlotRows(rows []Row, fileName string) error {
+func PlotRows(rows []Row, fileName string) (err error) {
+	bar := getStackedBarPlot(rows)
+	completePath, err := os.Getwd()
+	if err != nil {
+		panic(err)
```
Comment (krantideep95) [inline amortization.go:122]: return err

**Review by krantideep95 — APPROVED** [review]

**Pull request merged**